src/Widgets/QueryParamsDialog.vala

```diff
@@ -0,0 +1,181 @@
+        if (parse_result) {
+            try {
+                holder.set_value (parsed_value);
+            }
+            catch (GLib.Error ex) {
+                return false;
+            }
+            entry.get_style_context ().remove_class ("error");
+            return true;
+        }   
+        return false;
+    }
+
+    private bool set_value_for_holder (Gda.Holder holder, Gtk.Widget widget) {
+        GLib.Type holder_g_type = holder.get_g_type ();
+        if (holder_g_type == GLib.Type.BOOLEAN) {
+            Gtk.Switch switch = widget as Gtk.Switch;
+            if (switch == null) {
+                return false;
+            }
+            try {
+                holder.set_value (switch.get_active ());
+            }
+            catch (GLib.Error ex) {
+                return false;
+            }
+            return true;
+        }
+        else {
+            Gtk.Entry entry = widget as Gtk.Entry;
+            string text = entry.get_text ();
+
+            bool parse_result = false;
+            GLib.Value parsed_value;
+
+            if (holder_g_type == GLib.Type.INT) {
+                parse_result = int.try_parse (text, out parsed_value);
+            }
+            else if (holder_g_type == GLib.Type.UINT) {
+                parse_result = uint.try_parse (text, out parsed_value);
+            }
+            else if (holder_g_type == GLib.Type.FLOAT) {
+                parse_result = float.try_parse (text, out parsed_value);
+            }
+            else if (holder_g_type == GLib.Type.DOUBLE) {
+                parse_result = double.try_parse (text, out parsed_value);
+            }
+            else {
+                parse_result = true;
+                parsed_value = text;
+            }
+            return store_parsed_value (parse_result, parsed_value, holder, entry);
+        }
+    }
+
+    private void build_content () {
+        var content = get_content_area ();
+
+        var form_grid = new Gtk.Grid ();
+        form_grid.margin = 15;
+        form_grid.row_spacing = 12;
+        form_grid.column_spacing = 20;
+
+        entries = new Gee.ArrayList<Gtk.Widget> ();
+
+        for (int i = 0; ; i++) {
+            Gda.Holder? holder = params.get_nth_holder (i);
+            if (holder == null) {
+                break;
+            }
+            var holder_id = holder.get_id ();
+
+            var label = new Gtk.Label (holder_id);
+            form_grid.attach (label, 0, i, 1, 1);
+            var entry = entry_for_holder (holder);
+            form_grid.attach (entry, 1, i, 1, 1);
+            entries.add (entry);
+        }
+
+        content.add (form_grid);
+
+        var run_button = new Sequeler.Widgets.RunQueryButton ();
+        run_button.set_sensitive (true);
+        add_action_widget (run_button, Action.RUN_QUERY);
+
+        // TODO: cancel_button should keep a consistent style
+        var cancel_button = new Sequeler.Widgets.CancelQueryButton ();
```

Comment:
Same thing here for the run query button.
No need to add a dedicated class if this is not repeated or reused.

Reply:
Moved to `build_cancel_button`